packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -196,7 +196,7 @@ const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selec
     <DndProvider backend={HTML5Backend}>
       <Context.Provider
         value={{
-          rootNodes: rootNodes.map(toSceneHeirarchyNode),
+          rootNodes: rootNodes.map((node) => toSceneHeirarchyNode(node, !isEmpty(searchTerms))),
```

Comment:
Looking at this, I would almost prefer to see a "filteredNodes" property passed through context instead, but we can save that for when we rework search properly.

Reply:
a filter callback? that's probably more scalable. i can have a try in my next change